srcs/methods/GETMethod.cpp

```diff
@@ -15,22 +17,47 @@ namespace Webserver
 
 	}
 
-	// add !isMethodAllowed()
+
+	static bool	endsOnSlash(const std::string& target)
+	{
+		const size_t lastChar = target.size() - 1;
+		
+		return target[lastChar] == '/';
+	}
+
+	static bool	dirHasDefaultIndex(const std::string path)
+	{
+		struct stat fileInfo;
+
+  		return stat(path.c_str(), &fileInfo) == 0; 
+	}
+
 	Response* GETMethod::process(const std::string& uri)
 	{
 		DEBUG("Entering GET method!");
 
 		std::ifstream*	stream = new std::ifstream();
-		std::string		target = _request.getTarget();
+		std::string		target(uri);
 
-		if (target == "/")
-			target = _host.getDefaultIndex();
-		stream->open(prependRoot(_host.getRoot(), target));
+		if (endsOnSlash(target))
+		{
+			if (_host.getDefaultIndex() != "" && dirHasDefaultIndex(target + _host.getDefaultIndex()))
+				target = uri + _host.getDefaultIndex();
+			else if (_host.isAutoIndexEnabled())
+			{
+				std::stringstream *oss = new std::stringstream();
+				*oss << AutoIndex(target).getHtmlPage();
+				return new OkStatusResponse(oss, HttpStatusCodes::OK);
```

Comment:
Is fine, but could also be 
`return new AutoIndexResponse(AutoIndex(target).getHtmlpage());`